src/ecdsa.js

```diff
@@ -9,9 +9,36 @@ var ZERO = new Buffer([0])
 var ONE = new Buffer([1])
 
 // https://tools.ietf.org/html/rfc6979#section-3.2
-function deterministicGenerateK(curve, hash, d) {
+function deterministicGenerateK(curve, hash, d, checkSig) {
   typeForce('Buffer', hash)
   typeForce('BigInteger', d)
+//  typeForce('Function', checkSig)
+
+  // FIXME: remove in 2.0.0
+  if (typeof checkSig !== 'function') {
+    console.warn('deterministicGenerateK requires a checkSig callback in 2.0.0, see #337 for more information')
+
+    checkSig = function(k) {
```

Comment:
Definitely not worth any refactoring effort.
I assume you have no other issues with this PR?
On 8 Jan 2015 11:34 pm, "Wei Lu" <notifications@github.com> wrote:

> In src/ecdsa.js
> <https://github.com/bitcoinjs/bitcoinjs-lib/pull/337#discussion-diff-22649106>
> :
>
> >    typeForce('Buffer', hash)
> >    typeForce('BigInteger', d)
> > +//  typeForce('Function', checkSig)
> > +
> > +  // FIXME: remove in 2.0.0
> > +  if (typeof checkSig !== 'function') {
> > +    console.warn('deterministicGenerateK requires a checkSig callback in 2.0.0, see #337 for more information')
> > +
> > +    checkSig = function(k) {
>
> It's not ideal that this function is repeated here with minor
> modification. I'm not if this is an abstraction problem as the function
> below used in sign has the side effect of modifying s and r. Maybe it's
> not worth the effort to refactor this as it will be removed in 2.0.0.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitcoinjs/bitcoinjs-lib/pull/337/files#r22649106>.
>